src/renderer/components/table/react-table.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import styles from "./react-table.module.css";
+import React from "react";
+import { useCallback, useMemo } from "react";
+import { useFlexLayout, useSortBy, useTable, UseTableOptions } from "react-table";
+import { Icon } from "../icon";
+import { cssNames } from "../../utils";
+
+interface Props extends UseTableOptions<any> {
+  headless?: boolean;
+}
+
+export function ReactTable({ columns, data, headless }: Props) {
+  const defaultColumn = useMemo(
+    () => ({
+      minWidth: 20,
+      width: 100,
+    }),
+    []
+  );
```

Comment:
[nit] I think it's not needed to wrapped with `useMemo()` (as they are static value in every re-render) but no harm if wrapped.
```suggestion
  const defaultColumn = {
      minWidth: 20,
      width: 100,
  };
```

Reply:
Yes, I guess it can be removed. I've used the "recommendations" from the `react-table` author. There are many things memorized in there https://react-table.tanstack.com/docs/examples/virtualized-rows